src/cargo/ops/cargo_remove.rs

```diff
@@ -46,20 +58,118 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
 
         manifest.remove_from_table(&dep_table, dep)?;
 
-        // Now that we have removed the crate, if that was the last reference to that
-        // crate, then we need to drop any explicitly activated features on
-        // that crate.
+        // Now that we have removed the crate, if that was the last reference to that crate, then we
+        // need to drop any explicitly activated features on that crate.
         manifest.gc_dep(dep);
     }
 
+    // remove from workspace dependencies
+
+    let mut ws_manifest = None;
+
+    if !to_remove_workspace.is_empty() {
+        // read the packages in the workspace, substituting in the modified manifest
+        let members = options
+            .workspace
+            .members()
+            .map(|p| {
+                if p.manifest_path() == manifest_path {
+                    Ok((p, manifest.clone()))
+                } else {
+                    LocalManifest::try_new(p.manifest_path()).map(|m| (p, m))
+                }
+            })
+            .collect::<CargoResult<Vec<_>>>()?;
+
+        if options.workspace.root_manifest() != manifest_path {
+            let manifest: toml_edit::Document =
+                cargo_util::paths::read(options.workspace.root_manifest())?.parse()?;
+            ws_manifest = Some(manifest);
+        }
+
+        let ws_manifest = match &mut ws_manifest {
+            Some(manifest) => manifest,
+            None => &mut manifest.data,
+        };
+
+        for dep in to_remove_workspace {
+            if !dep_in_workspace(dep, &members) {
+                remove_workspace_dep(dep, ws_manifest);
+                options
+                    .config
+                    .shell()
+                    .status("Removing", format!("{dep} from workspace dependencies"))?;
+            }
+        }
+    }
+
     if options.dry_run {
         options
             .config
             .shell()
             .warn("aborting remove due to dry run")?;
     } else {
         manifest.write()?;
+
+        if let Some(ws_manifest) = ws_manifest {
+            cargo_util::paths::write(
+                options.workspace.root_manifest(),
+                ws_manifest.to_string().as_bytes(),
+            )?;
+        }
```

Comment:
This has aliasing issues: if these point to the same file, last writer wins and won't pick up changes from the other.

Reply:
Oh, you do tricks to catch if you are editing the root package.  Again, this is probably not worth it.  The extra overhead / loss of atomicity is probably not worth the complexity and extra risk in refactors causing aliasing issues

Reply:
Okay, I added a surrounding conditional to ensure that aliasing issues will not occur. I kept the "trick" from earlier so that duplicate reads will not occur though.

Reply:
I feel like simplifying the code (removing aliasing complete) would be the safest, simplest route for resolving this.  By adding in protection against aliasing that doesn't occur, developers will think even more that aliasing may occur and be wrong.  This protection wouldn't even be correct if an aliasing bug occurs because changes would be made in tow separate manifest objects.

Reply:
I'm not sure what the simplification you're thinking of would be

Reply:
What you had before, where you wrote the selected manifest before doing workspace GC.  You could even wrap it in a scope / explicitly drop the selected manifest to ensure any access to it later is a compiler error.

Reply:
Okay, done! I'm much happier with this solution.